pytest_trio/plugin.py

```diff
@@ -354,29 +354,8 @@ def _trio_test_runner_factory(item, testfunc=None):
         testfunc = item.obj
 
         config_run = choose_run(config=item.config)
-        runs = {
-            marker.kwargs.get('run', config_run)
-            for marker in item.iter_markers("trio")
-        }
-
-        if len(runs) == 0:  # pragma: no cover
-            raise RuntimeError(
-                "No 'trio' marker found.  Please report to pytest-trio."
-            )
-        elif len(runs) == 1:
-            [run] = runs
-        else:
-            runs.discard(trio.run)
-            if len(runs) == 1:
-                [run] = runs
-            else:
-                runs_string = ', '.join(
-                    sorted(f"{f.__module__}.{f.__qualname__}" for f in runs)
-                )
-                raise ValueError(
-                    f"Not yet able to select from more than one third-party"
-                    f" runner.  Found: {runs_string}"
-                )
+        marker = item.get_closest_marker("trio")
```

Comment:
I think you probably still want to iterate through all the markers and pick the closest one with an explicitly specified `run` kwarg? That way an outer `pytest.mark.trio(run=xyz)` won't overshadow an inner `pytest.mark.trio` without arguments.

Reply:
I'm probably misreading but those two comments seem to conflict.  Are you suggesting that the closest marker with `run` specified should be used?  Or that the closest marker regardless should be used?  I had thought the plan was just to get the closest marker and whatever it said goes.  If no `run` is specified right there then it gets to be a generic Trio test.

Reply:
My intuition here is: find the narrowest-scoped place where someone specified a runner, and use that, falling back to the config file if no trio marker specifies a runner. Markers that don't specify a runner don't contribute any useful information about what runner is desired, because if you didn't `enable_trio_mode` then you might need to say `@pytest.mark.trio` on *all* your tests, and you should still be able to make the default be qtrio in the config file.